src/runtime/refiner.ts

```diff
@@ -404,6 +420,30 @@ export class BinaryExpression extends RefinementExpression {
           }
           return this;
       }
+      case Op.DIV: {
+          if (this.rightExpr instanceof NumberPrimitive && this.rightExpr.value === 1) {
+              return this.leftExpr;
+          }
+          if (this.rightExpr instanceof BigIntPrimitive && this.rightExpr.value === BigInt(1)) {
+              return this.leftExpr;
+          }
+          return this;
+      }
+      case Op.MUL: {
+          if (this.leftExpr instanceof NumberPrimitive && this.leftExpr.value === 1) {
+              return this.rightExpr;
+          }
+          if (this.leftExpr instanceof BigIntPrimitive && this.leftExpr.value === BigInt(1)) {
+              return this.rightExpr;
+          }
+          if (this.rightExpr instanceof NumberPrimitive && this.rightExpr.value === 1) {
+              return this.leftExpr;
+          }
+          if (this.rightExpr instanceof BigIntPrimitive && this.rightExpr.value === BigInt(1)) {
+              return this.leftExpr;
+          }
+          return this;
+      }
```

Comment:
Are Numbers and BigInts going to be interoperable?

Reply:
I'm hoping to give literals the ability to be either Number or BigInt backed (based on the presence of a decimal point e.g. 2.0 is a Number, 2 can be either Number or BigInt and as a BigInt it can be compared with and added to any BigInt backed type, so Int, Long etc).

This is to avoid having to have `2n` for BigInt, `2L` for Long, `2i` for Int etc.
I don't think there's any way to operate/cast floating and integer types that is definitely safe, but we could support it if we make the semantics well known.